core/src/Cuda/Kokkos_Cuda_Instance.hpp

```diff
@@ -117,6 +119,10 @@ class CudaInternal {
   uint32_t* m_scratchConcurrentBitset;
   cudaStream_t m_stream;
 
+  // Team Scratch Level 1 Space
+  mutable int64_t m_team_scratch_current_size;
+  mutable void* m_team_scratch_ptr;
```

Comment:
Why did you declare them mutable?

`Cuda:: impl_internal_space_instance` returns a pointer to non-const*.  So I don't think you need it in principle.
https://github.com/kokkos/kokkos/blob/c01580a271257704a50171e6527cd1f36155496e/core/src/Kokkos_Cuda.hpp#L254-L256

*Whether it should is another question 🙄

Reply:
Just talking to Damien: this is interesting problem but maybe a bigger one? We can either get rid of the mutable and leave it a non-const pointer or leave the mutable and make this a const pointer. What is the best pattern here? @nliber

Reply:
maybe we leave both of them for now and come back to this later.